diff -Naur ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu.c ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu.c
--- ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu.c	2024-07-19 04:57:58.000000000 +0200
+++ ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu.c	2024-08-06 08:05:26.856852351 +0200
@@ -15,6 +15,7 @@
 #include <linux/timer.h>
 #include <linux/sched.h>
 #include <linux/version.h>
+#include <linux/vmalloc.h>
 
 #include "ipu.h"
 #include "ipu-buttress.h"
diff -Naur ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu-mmu.c ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu-mmu.c
--- ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu-mmu.c	2024-07-19 04:57:58.000000000 +0200
+++ ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu-mmu.c	2024-08-06 08:05:26.856852351 +0200
@@ -7,6 +7,7 @@
 #include <linux/iova.h>
 #include <linux/module.h>
 #include <linux/sizes.h>
+#include <linux/vmalloc.h>
 
 #include "ipu.h"
 #include "ipu-platform.h"
diff -Naur ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu-psys.c ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu-psys.c
--- ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462.old/drivers/media/pci/intel/ipu-psys.c	2024-07-19 04:57:58.000000000 +0200
+++ ipu6-drivers-9369b88ec1ea03670fb2dbfe7abdff411683d462/drivers/media/pci/intel/ipu-psys.c	2024-08-06 08:05:26.856852351 +0200
@@ -258,6 +258,22 @@
 	return NULL;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 10, 0)
+/* Dropped from 6.10, use our own copy for now */
+static int follow_pfn(struct vm_area_struct *vma, unsigned long address, unsigned long *pfn)
+{
+	spinlock_t *ptl;
+	pte_t *ptep;
+
+	if (follow_pte(vma, address, &ptep, &ptl))
+		return -EINVAL;
+
+	*pfn = pte_pfn(ptep_get(ptep));
+	pte_unmap_unlock(ptep, ptl);
+	return 0;
+}
+#endif
+
 static int ipu_psys_get_userpages(struct ipu_dma_buf_attach *attach)
 {
 	struct vm_area_struct *vma;
