--- ivsc-driver-b5969f9311c07a80250c3ab5e1174a792195e8e3/drivers/misc/mei/hw-vsc.c	2024-07-04 12:16:58.202931055 +0200
+++ ivsc-driver-b5969f9311c07a80250c3ab5e1174a792195e8e3/drivers/misc/mei/hw-vsc.c	2024-07-04 12:17:08.510034125 +0200
@@ -339,14 +339,9 @@
 /* %s is sensor name, need to be get and format in runtime */
 static char *fw_name_template[][3] = {
 	{
-		"vsc/soc_a1/ivsc_fw_a1.bin",
-		"vsc/soc_a1/ivsc_pkg_%s_0_a1.bin",
-		"vsc/soc_a1/ivsc_skucfg_%s_0_1_a1.bin",
-	},
-	{
-		"vsc/soc_a1_prod/ivsc_fw_a1_prod.bin",
-		"vsc/soc_a1_prod/ivsc_pkg_%s_0_a1_prod.bin",
-		"vsc/soc_a1_prod/ivsc_skucfg_%s_0_1_a1_prod.bin",
+		"intel/vsc/ivsc_fw.bin",
+		"intel/vsc/ivsc_pkg_%s_0.bin",
+		"intel/vsc/ivsc_skucfg_%s_0_1.bin",
 	},
 };
 
